### ISSUE TITLE
Add documentation on starting the machine

### ISSUE THREAD

**Issue by bschlenk** [carloslfu/use-machine#5]
Took me a while to realize that calling useMachine does not start the machine. Is there a good reason to leave that up to the caller? The machine is stopped when unmounting, so maybe it makes sense to start it when it is first called? If not, then it would be nice to document this. FWIW, I had to add this code to start my machine:

```typescript
const machine = useMachine(...);

useEffect(() => {
  machine.service.start();
}, []);
```

**Comment by carloslfu** [comment]
Yes, you are right. I think it's better to start the machine. Thanks, @bschlenk .

**Comment by carloslfu** [comment]
Hi @bschlenk. No, you don't have to, the machine starts [here](https://github.com/carloslfu/use-machine/blob/master/src/index.ts#L40). Maybe you discovered other issue, can you share more details? See a working example [here](https://codesandbox.io/s/5z0820jlyk).

**Comment by bschlenk** [comment]
Weird, my machine doesn't seem to start unless I manually call start.

Here's where I'm calling `useMachine`: https://github.com/bschlenk/set/blob/master/src/state/useSetMachine.ts
And here's where I'm defining my state machine: https://github.com/bschlenk/set/blob/master/src/state/xstate.ts

Looking through the debugger, it looks like `machine.state.context` has the right values, but `machine.context` is stale. It might be because I have an `onEntry` action on the initial state that updates the context. Maybe `useMachine` isn't updating its internal `context` state?

**Comment by carloslfu** [comment]
I'll take a look.

**Comment by bschlenk** [comment]
I created a minimal example, but this may actually be an issue with xstate, it looks like it isn't calling `onChange` for actions triggered during initialization.

https://codesandbox.io/s/4zjv035mo9

I took your example and wrapped it in two outer states: `setup` and `main`. I also added a `name` field to the context that defaults to `"unset"`. Setup has an `onEntry` action that updates `name`, and then an immediate transition to `main`. If you look at the output, `name` doesn't get updated as expected. But if you uncomment the `machine.service.start` call on line 86, the state does update.

If you leave the `start` call commented and click one of the tick buttons, you can see that `name` does update. So the state machine is holding the correct state, it just isn't notifying react that the state context has changed during initialization.

**Comment by bschlenk** [comment]
I found the issue. `onTransition` and `onChange` are called *after* `init`, so anything that happens during initialization is missed. Calling `init` after both callbacks are set fixes the problem.
